upload.py

```diff
@@ -15,87 +16,175 @@
 
 
 def generate_nuclia_sync():
+    """
+    Generate a dictionary of HTML files and their MD5 hashes.
+
+    Args:
+        None
+
+    Returns:
+        dict: A dictionary of HTML files and their MD5 hashes.
+    """
     result = {"docs": {}}
-    for doc in glob.glob("./docs/**/*.md", recursive=True):
+    for doc in glob.glob("./_build/html/**/*.html", recursive=True):
+        
+        # Skip pages that don't need to be indexed
+        if any(skip in doc for skip in ['search.html', 'genindex.html','webpack-macros.html']):
+            continue
+
         hash = hashlib.md5(open(doc, "rb").read()).hexdigest()
         result["docs"][doc] = hash
     return result
 
 
-def extract_first_heading(file_path):
+def extract_content(file_path):
+    """
+    Extract title and content from HTML file using BeautifulSoup.
+
+    Args:
+        file_path (str): The path to the HTML file.
+    
+    Returns:
+        tuple: A tuple containing the title and content.
+    """
     with open(file_path, 'r', encoding='utf-8') as file:
-        content = file.read()
+        soup = BeautifulSoup(file.read(), 'html.parser')
 
-    # The pattern matches lines starting with one or more hash symbols followed by any non-empty text
-    pattern = r'^\s*#+\s+(.+)$'
-    match = re.search(pattern, content, re.MULTILINE)
+    # Extract title
+    title = soup.find('h1').get_text()
 
-    if match:
-        # Removing backticks
-        heading = match.group(1).replace('`', '').strip()
-        return heading
+    # Extract content
+    article = soup.find('article', class_='bd-article')
+    if not article:
+        return None, None
+    
+    return title, article
 
-    # If no heading is found
-    return None
 
+def get_slug(path):
+    """
+    Extract slug from path.
 
-# re.sub() is used to perform substitution
-# r'[\W_]+' matches any non-word characters (including underscores),and replaces them with a single hyphen '-'
-# strip('-') removes any leading or trailing hyphens from the resulting string
-get_slug = lambda path: re.sub(r'[\W_]+', '-', path.lower().strip()).strip('-')
+    Args:
+        path (str): The path to the HTML file.
+    
+    Returns:
+        str: The slug.
+    """
+    path = path.replace('./_build/html/', '')
+    path = re.sub(r'\.html$', '', path)
+    slug = re.sub(r'[\W_]+', '-', path.lower().strip()).strip('-')
+    return slug
 
 
 def normalize_path(path):
-    return path.replace('.md', '').replace('./docs', '').lstrip('/')
+    path = path.replace('./_build/html/', '')
+    return path.replace('.html', '').lstrip('/')
 
 
 def create_url(origin_url, url_path):
     return f"{origin_url.rstrip('/')}/{url_path}"
 
+def transform_path_to_url(path, base_url):

```

Comment:
I'm not sure we should do a transformation. Some trainings follow the `_` convention, others `-`. Is there a reason to do a transformation?